web/src/main/java/org/wso2/testgrid/web/api/TestPlanService.java

```diff
@@ -143,4 +148,39 @@ public Response createTestPlan(TestPlanRequest testPlanRequest) {
                             setDescription(e.getMessage()).build()).build();
         }
     }
+
+    /**
+     * Save the {@link TestPlanRequest} object as a YAML file.
+     * @param testPlanRequest TestPlanRequest needs to be saved.
+     * @throws TestGridException thrown when error occurred;
+     *              1.If file directory (which is not existing) can not be created.
+     *              2.If IOException occured while writing YAML file.
+     *              3.If the YAML file already exists.
+     */
+    private void persistAsYamlFile(TestPlanRequest testPlanRequest) throws TestGridException {
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+        String path = Constants.TESTPLANS_DIR + testPlanRequest.getTestPlanName();
+
+        File directory = new File(path);
+        File yamlFile = new File(path + File.separator + testPlanRequest.getTestPlanName() + ".yml");
+
+        if (!directory.exists()) {
+            if (!directory.mkdir()) {
```

Comment:
This will only create the child folder.. If you want the parent folders also created, then we need to specifically say so. ie. the equivalent of UNIX 'mkdir -p' argument.

Reply:
Fixed by moving to `java.nio.file.Files.createDirectories ` which creates all non existing directories in the given path.